handlers/svg-open-street-map/osm-svg.py

```diff
@@ -0,0 +1,170 @@
+        request_schema["$id"]: request_schema,
+        response_schema["$id"]: response_schema,
+        renderer_schema["$id"]: renderer_schema,
+    }
+    resolver = jsonschema.RefResolver.from_schema(
+        request_schema, store=store
+    )
+    # Get and validate the request contents
+    contents = request.get_json()
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema, resolver=resolver
+        )
+        validator.validate(contents)
+    except ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid request received!"), 400
+
+    preprocessor = contents['preprocessors']
+    # Check preprocessor data
+    if "ca.mcgill.a11y.image.preprocessor.openstreetmap"\
+            not in preprocessor:
+        logging.info("Not for OSM preprocessor. Skipping ...")
+        return "", 204
+
+    if "dimensions" in contents:
+        # If an existing graphic exists, often it is
+        # best to use that for convenience.
+        # see the following for SVG coordinate info:
+        # developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Positions
+        dimensions = contents["dimensions"]
+    else:
+        # Use default dimensions
+        dimensions = 500, 500
+    svg = draw.Drawing(dimensions[0], dimensions[1])
+    svg_layers = []
+    data = preprocessor["ca.mcgill.a11y.image.preprocessor.openstreetmap"]
+    if "streets" in data:
+        streets = data["streets"]
+        lat = data["bounds"]["latitude"]
+        lon = data["bounds"]["longitude"]
+        lon_min = lon["min"]
+        lat_min = lat["min"]
+        lon_max = lon["max"]
+        lat_max = lat["max"]
+        # Scale the lon/lat units to svg pixels equivalent.
+        scaled_longitude = dimensions[0] / (lon_max - lon_min)
+        scaled_latitude = dimensions[1] / (lat_max - lat_min)
+        bounds = [[lon_min, lat_min],
+                  [lon_min, lat_max],
+                  [lon_max, lat_max],
+                  [lon_max, lat_min],
+                  [lon_min, lat_min]]
+        # Draw bounding box for the streets
+        for index in range(len(bounds) - 1):
+            p = draw.Path(stroke="orange", stroke_width=6, fill='none')
+            p.M(scaled_longitude * (bounds[index][0] - lon_min),
+                scaled_latitude * (bounds[index][1] - lat_min))
+            p.L(scaled_longitude * (bounds[index + 1][0] - lon_min),
+                scaled_latitude * (bounds[index + 1][1] - lat_min))
+            svg.append(p)
+        # Latitude is north-south, Longitude is east-west
+        for street in range(len(streets)):
+            foo = object()
+            p = draw.Path(
+                stroke=Color(
+                    pick_for=foo),
+                stroke_width=1.5,
+                fill='none')
+            node_coordinates = [[node["lon"], node["lat"]]
+                                for node in streets[street]["nodes"]]
+            for index in range(len(node_coordinates) - 1):
```

Comment:
Why not iterate directly on `node_coordinates`?

Reply:
Because I needed the current and the next list items at the same time. I don't know if there are better options.